public/app/plugins/datasource/prometheus/datasource.test.ts

```diff
@@ -94,6 +94,16 @@ describe('PrometheusDatasource', () => {
     ds = new PrometheusDatasource(instanceSettings, templateSrvStub, timeSrvStub);
   });
 
+  // Some functions are required by the parent datasource class to provide functionality such as ad-hoc filters, which requires the definition of the getTagKeys, and getTagValues functions
```

Comment:
That's a good comment. I was gonna remove them here https://github.com/grafana/grafana/pull/66466/files#diff-b6a6838d619292fa468c692a4a246b033da56d8dd5d4cad9d47a549345f59780L557-L568

Can you point me where exactly those are being called?

Reply:
[AdHocFilterKey.tsx](https://github.com/grafana/grafana/blob/main/public/app/features/variables/adhoc/picker/AdHocFilterKey.tsx)
```
if (!ds || !ds.getTagKeys) {
    return [];
  }
```